### DISCUSSION TITLE
Change in glyph positioning

### DISCUSSION THREAD

**Discussion by pahujar** [harfbuzz/harfbuzz#3525] [General] [answered]
Hi,
We have upgraded harfbuzz from version 2.74 to 3.3.2 and are seeing slight difference in glyph advances when retrieved using API 
hb_font_set_scale(pHbFont,  +from_float(x_scale),  -from_float(y_scale));
hb_glyph_position_t* pos = hb_buffer_get_glyph_positions(pBuffer, NULL);

the difference is in 0.000xxx decimal place.

was there any change in this regard ? Any help is much appreciated ?
Thanks.

**Comment by behdad** [comment, marked as answer]
Yes there was a small fix in how we round glyphs, which unfortunately had this side effect.

This was fixed in #3274 bbaccf23d9e231370fb3ca3398ebfc132c7555bc

**Comment by pahujar** [comment]
Thanks for replying.

The version we are using i.e. 3.3.2, seems to have this change already. Probably that change explains the difference we are seeing here. Does it need improvements ? 
e.g. The difference we are seeing is that x-advance value for glyph  (font is arial bold, happening with many other fonts & glyphs too)
- in version 2.7.2 was -  98296
- in version 3.3.2 is - 98297

when converted these values to float, there is difference in decimal places. Any help is appreciated.
Thanks.

**Comment by behdad** [comment]
As I said, this was a result of a rounding change and as such intended.
